align_system/algorithms/multi_comparison_adm.py

```diff
@@ -0,0 +1,120 @@
+import json
+import yaml
+import os
+
+from align_system.algorithms.lib.aligned_decision_maker import AlignedDecisionMaker
+from align_system.algorithms.lib.chat.chat_language_model import ChatLanguageModel
+
+
+class Option:
+    def __init__(self, model, scenario, probe, choice, characteristic, definition, template, log_file=None):
+        self.model = model
+        self.choice = choice
+        self.scenario = scenario
+        self.probe = probe
+        self.characteristic = characteristic
+        self.definition = definition
+        self.template = template
+        self.log_file = log_file
+    
+    def __lt__(self, other):
+        assert self.scenario == other.scenario
+        assert self.probe == other.probe
+        
+        for _ in range(10):
+            response = self.model.generate_from_template(self.template, substitution_dicts={
+                'scenario': self.scenario,
+                'probe': self.probe,
+                'characteristic': self.characteristic,
+                'definition': self.definition,
+                'option_0': self.choice,
+                'option_1': other.choice
+            }, log_file=self.log_file)[0]
+            
+            try:
+                response = json.loads(response)
+                assert 'decision' in response
+                break
+            except json.JSONDecodeError:
+                continue
+            
+        return response['decision'] == 1
+
+
+class MultiComparisonADM(ChatLanguageModel, AlignedDecisionMaker):
+    
+    def predict_kdma_values(self, sample, kdmas, **kwargs):
+        scenario = sample['scenario']
+        probe = sample['probe']
+        choices = sample['choices']
+        
+        kdma_descriptions_file = kwargs['kdma_descriptions_file']
+        kdma_descriptions_file = os.path.join(os.path.dirname(__file__), kdma_descriptions_file)
+        # read yaml file
+        with open(kdma_descriptions_file) as f:
+            kdma_descriptions = yaml.load(f, Loader=yaml.FullLoader)
+            
+        comparison_template_file = kwargs['comparison_template_file']
+        # make the path relative to the current file
+        comparison_template_file = os.path.join(os.path.dirname(__file__), comparison_template_file)
+        with open(comparison_template_file) as f:
+            template = f.read()
+        
+        n = len(choices)
+        points = [round(point/100) for point in range(0, 1000 + int(1000/n), int(1000/(n-1)))]
```

Comment:
Please add a comment explaining what points is here, and some example values for `n` choices (e.g. 4)